src/llm/apis/openai_completions.cpp

```diff
@@ -874,6 +874,89 @@ std::string OpenAIChatCompletionsHandler::serializeUnaryResponse(const ov::genai
     return buffer.GetString();
 }
 
+std::string OpenAIChatCompletionsHandler::serializeUnaryResponse(const ov::genai::VLMDecodedResults& results) {  // TODO separate common part with function implemented above
```

Comment:
I have already seen such code in last PR 🤔 
each combination of VLM/TEXT x Legacy/CB has its own type we must deserialize from...?

Reply:
I think the best solution would be to make changes in GenAI, so we can handle this without such logic (we could also have some additional data added to the output - like logprobs and finish reason that are missing now). But this will not happen in 2025.1